samples/cellular/modem_shell/src/link/link_shell.c

```diff
@@ -2664,6 +2722,10 @@ SHELL_STATIC_SUBCMD_SET_CREATE(
 		tau, NULL,
 		"Subscribe/unsubscribe for modem TAU pre-warning notifications.",
 		link_shell_tau, 0, 10),
+	SHELL_CMD_ARG(
+		modem, NULL,
+		"Initialize and shutdown modem.",
+		link_shell_modem, 2, 0),
```

Comment:
`link modem` without options causes an error, not help being printed:
```
mosh:~$ link modem
modem: wrong parameter count
modem - Initialize and shutdown modem.
```
I think it would be better to handle this command in the same way as others, so parameters should be optional and validity checked during parsing.

Reply:
I added a comment about this before reading this. You could remove the `while()` loop, which would make the parsing simpler. And if there was a `return` after each valid option, there would be no need for the `operation_selected` flag, I guess. Unless you want to enable the behavior you describe above. I think this feels like a bug, because other commands behave differently.

Reply:
The first question is that do we specifically want to block this:
`link modem --shutdown --init`

If yes, I'll remove the while loop.

This command is bit different because it just has the operation and no operation specific sub-parameters so above is something we may intentionally leave as a possibility.
E.g., PSM and eDRX has `-e` and `-d` and `-e` has sub-parameters so this wouldn't work easily.

Reply:
We've decided to leave the option of setting multiple options.